### DISCUSSION TITLE
white-box attack for Preprocessor class

### DISCUSSION THREAD

**Discussion by le000043** [Trusted-AI/adversarial-robustness-toolbox#1484] [Q&A] [answered]
Hello,
Thank you for creating this wonderful software. I see that in the file examples/get_started_inverse_gan.py, the FGSM attack was conducted in a gray-box setting, which means the attacker happens before the defense is applied. So my question is how can I make the attacker be aware of the preprocessor/defender InverseGAN? I suppose that I need to implement a Wrapper class? If so, what functions do I have to override?

**Comment by beat-buesser** [comment, marked as answer]
Hi @le000043 That's true. You would have to modify the script to pass `inverse_gan` to argument `preprocessing_defences` of a new instance of `TensorFlowClassifier` defined similarly as in line 46. Does that work?
